src/main/scala/is/hail/asm4s/FunctionBuilder.scala

```diff
@@ -214,6 +214,7 @@ class FunctionBuilder[F >: Null](parameterTypeInfo: Array[MaybeGenericTypeInfo[_
         try {
           if (f == null) {
             this.synchronized {
+              initf()
               if (f == null) {
```

Comment:
This could execute `initf` twice. Consider two threads. `f` starts `null`.

```
thread 1                  thread 2                    value of f
if (f == null)            idle                        null

idle                      if (f == null)              null

idle                      acquire lock on `this`      null

fail to acquire                                       null
lock because 2
already has it

idle                      if (f == null)              null
idle                      initf()                     null
idle                      f = ...                     null
idle                      release lock on `this`      <some object>
acquire lock on `this`    idle                        <some object>
initf()                   idle                        <some object>    // double call to initf!
if (f == null)            idle                        <some object>
release lock on `this`    idle                        <some object>
return from `apply`       idle                        <some object>
idle                      return from `apply`         <some object>
```

Moving `initf` into the `if`-statement is sufficient to avoid this issue.